src/main/java/ch/njol/skript/effects/EffBroadcast.java

```diff
@@ -65,56 +75,96 @@ public class EffBroadcast extends Effect {
 		List<CommandSender> receivers = new ArrayList<>();
 		if (worlds == null) {
 			receivers.addAll(Bukkit.getOnlinePlayers());
 			receivers.add(Bukkit.getConsoleSender());
 		} else {
-			for (World world : worlds.getArray(e))
+			for (World world : worlds.getArray(event))
 				receivers.addAll(world.getPlayers());
 		}
 
 		for (Expression<?> message : getMessages()) {
 			if (message instanceof VariableString) {
-				BaseComponent[] components = BungeeConverter.convert(((VariableString) message).getMessageComponents(e));
+				if (!dispatchEvent(getRawString(event, (VariableString) message), receivers))
+					continue;
+				BaseComponent[] components = BungeeConverter.convert(((VariableString) message).getMessageComponents(event));
 				receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
 			} else if (message instanceof ExprColoured && ((ExprColoured) message).isUnsafeFormat()) { // Manually marked as trusted
-				for (Object realMessage : message.getArray(e)) {
+				for (Object realMessage : message.getArray(event)) {
+					if (!dispatchEvent(Utils.replaceChatStyles((String) realMessage), receivers))
+						continue;
 					BaseComponent[] components = BungeeConverter.convert(ChatMessages.parse((String) realMessage));
 					receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
 				}
 			} else {
-				for (Object messageObject : message.getArray(e)) {
+				for (Object messageObject : message.getArray(event)) {
 					String realMessage = messageObject instanceof String ? (String) messageObject : Classes.toString(messageObject);
+					if (!dispatchEvent(Utils.replaceChatStyles(realMessage), receivers))
+						continue;
 					receivers.forEach(receiver -> receiver.sendMessage(realMessage));
 				}
 			}
 		}
 	}
 
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "broadcast " + messageExpr.toString(event, debug) + (worlds == null ? "" : " to " + worlds.toString(event, debug));
+	}
+
 	private Expression<?>[] getMessages() {
 		if (messageExpr instanceof ExpressionList && !messageExpr.getAnd()) {
 			return new Expression[] {CollectionUtils.getRandom(messages)};
 		}
 		return messages;
 	}
 
-	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return "broadcast " + messageExpr.toString(e, debug) + (worlds == null ? "" : " to " + worlds.toString(e, debug));
+	/**
+	 * Manually calls a {@link BroadcastMessageEvent}.
+	 * @param message the message
+	 * @return true if the dispatched event does not get cancelled
+	 */
+	@SuppressWarnings("deprecation")
+	private static boolean dispatchEvent(String message, List<CommandSender> receivers) {
+		Set<CommandSender> recipients = Collections.unmodifiableSet(new HashSet<>(receivers));
+		BroadcastMessageEvent broadcastEvent;
+		if (!Skript.isRunningMinecraft(1, 13)) {
+			broadcastEvent = new BroadcastMessageEvent(!Bukkit.isPrimaryThread(), message, recipients);
+		} else {
```

Comment:
This will always be false

Reply:
1.14 onwards the event can be called async, where you cannot specify synchronous state in earlier versions (1.13 for our case) https://github.com/SkriptLang/Skript/pull/5979/commits/88dee31f983be5a2c8e7b6b7710b89288bbf10ec

Reply:
Oh, I see the issue now at the 3/4/5th glance, oops.